File: 027_RemoveElement.py
class Solution(object):

    def removeElement(self, nums, val):
        """
        :type nums: List[int]
        :type val: int
        :rtype: int
        """

        pos = 0
        for i in xrange(len(nums)):
            if nums[i]==val:
                continue
            else:
                nums[pos] = nums[i]
                pos+=1

        return pos

if __name__ == '__main__':

    list = [1,2,8,2,4,6,8,8,10,1]
    print 'input list :',list, '  len is ',len(list)

    k = Solution()
    r = k.removeElement(list,8)

    print 'output list is ',list,'  len  is ',r

File: 027_RemoveElement.js
/**
 * @param {number[]} nums
 * @param {number} val
 * @return {number}
 */
var removeElement = function(nums, val) {
    if(nums.length===0){
        return(0);
    };
    var i = 0,end=0;
    for(i=0;i<nums.length;i++){
        if(nums[i]!==val){
            nums[end]=nums[i];
            end++;
        };

    };
    return(end);
};

var string_example =[0,1,2,2,3,0,4,2];
var value = 2;
console.log(removeElement(string_example,value));
console.log(string_example);
console.log('-----------the end!----------------');